src/app/app.module.ts

```diff
@@ -53,6 +53,11 @@ import { SharingModalComponent } from './shared/sharing-modal/sharing-modal.comp
 //============
 
 const appRoutes: Routes = [
+  {
+    path: ':langId/:bookId',
+    // Redirecting old URL format to the new one
+    redirectTo: ':langId/tool/v1/:bookId'
```

Comment:
I'm not familiar with the Angular router, but does this route need `pathMatch: 'full'`? Without it it seems like a `:langId/:bookId/:page` route would match both this route and the one below. I'm not sure if that will cause issues, just pointing it out.